docs/source/extending/extending_models.rst

```diff
@@ -0,0 +1,240 @@
+Extending the Interaction Models
+================================
+Let's assume you have invented a new interaction model,
+e.g. this variant of :class:`pykeen.models.DistMult`
+
+.. math::
+
+    f(h, r, t) = <h, \sigma(r), t>
+
+where :math:`h,r,t \in \mathbb{R}^d`, and :math:`\sigma` denotes the logistic sigmoid.
+
+Picking a Base Class
+--------------------
+From the reference documentation on base models (:mod:`pykeen.models.base`), we can see that
+:class:`pykeen.models.base.EntityRelationEmbeddingModel` is a good candidate for a base class
+since we want to have embeddings for entities *and* relations.
+
+Implementing ``score_hrt()``
+----------------------------
+The only implementation we have to provide is of the `score_hrt` member function:
+
+.. code-block:: python
+
+    from pykeen.models.base import EntityRelationEmbeddingModel
+
+    class ModifiedDistMult(EntityRelationEmbeddingModel):
+        def score_hrt(self, hrt_batch):
+            # Get embeddings
+            h = self.entity_embeddings(  hrt_batch[:, 0])
+            r = self.relation_embeddings(hrt_batch[:, 1])
+            t = self.entity_embeddings(  hrt_batch[:, 2])
+            # evaluate interaction function
+            return h * r.sigmoid() * t
+
+The ``entity_embeddings`` and ``relation_embeddings`` are available for all
+:class:`pykeen.models.base.EntityRelationEmbeddingModel` and are instances of
+:class:`torch.nn.Embedding`.
+
+The ``hrt_batch`` is a long tensor representing the internal indices of the edges.
+The above example shows a very common way of slicing it to get separate lists of
+head indices (``hrt_batch[:, 0]``), relation indices (``hrt_batch[:, 1]``), and
+tail indices (``hrt_batch[:, 2]``). Then, they are passed to the embeddings to
+look up the actual values. This is vectorized, so the results are also lists
+of embeddings (2-tensors) on which vectorized math can be applied.
```

Comment:
Maybe replace lists with tensors? 
"This is vectorized, so the results are also **tensors** of embeddings"

Reply:
Fixed in a6afff03